16 Lazy Loading/src/index.tsx

```diff
@@ -2,19 +2,41 @@ import * as React from 'react';
 import * as ReactDOM from 'react-dom';
 import { Router, Route, IndexRoute, hashHistory } from 'react-router';
 import App from './components/app.tsx';
-import AboutPage from './components/about/aboutPage';
-import MembersPage from './components/members/membersPage';
-import MemberPage from './components/member/memberPage';
 
+//Loading single component in one chunk
+const lazyLoadAboutComponent = () => {
+  return {
+      getComponent: (location, callback)=> {
+        require.ensure([], require => {
+          callback(null, require('./components/about/aboutPage')["default"]);
+        }, 'AboutPage');
+      }
+    }
+};
+
+//Loading group of components in one chunk
```

Comment:
Two questions here:

- Could we just  have to functions lazyLoadMemberComponent and lazyLoadMembersComponent that just reffer to the same chunk ' MemberComponet' (probably the common function approach is cleaner but just to know if it's possible).

- On the other hand if we pass 'member' 'members', we could call the parameter pageName, the we could have something like:

```` 
const lazyLoadMemberAreaPage = (pageName) => {
  return {
      getComponent: (location, callback) => {
        require.ensure(['./components/member/memberPage', './components/members/membersPage'], require => {
          switch(memberPageName=) 
          {
             case 'member':
                callback(null, require('./components/member/memberPage')["default"]);
             break;
             case 'members':
                 callback(null, require('./components/members/membersPage')["default"]);
             break;
          }
        }, 'MemberComponents');
      }
    }
````